command/state_mv.go

```diff
@@ -287,22 +289,17 @@ func (c *StateMvCommand) Run(args []string) int {
 					// If we're moving to an address without an index then that
 					// suggests the user's intent is to establish both the
 					// resource and the instance at the same time (since the
-					// address covers both), but if there's an index in the
-					// target then the resource must already exist.
+					// address covers both). If there's an index in the
+					// target then allow creating the new instance here,
+					// inferring the mode from how the new address was parsed.
 					if addrTo.Resource.Key != addrs.NoKey {
-						diags = diags.Append(tfdiags.Sourceless(
-							tfdiags.Error,
-							msgInvalidTarget,
-							fmt.Sprintf("Cannot move to %s: %s does not exist in the current state.", addrTo, addrTo.ContainingResource()),
-						))
-						c.showDiagnostics(diags)
-						return 1
+						fromEachMode = eachModeForInstanceKey(addrTo.Resource.Key)
```

Comment:
Could/should we use the configuration as a safety check here to make sure the user is asking for something sensible? By which I mean, if there's no configuration block at the target resource at all then maybe we still fail, and likewise if the set-ness of the `count` or `for_each` of that resource disagrees with the selected each mode we could fail that too.

That would help make sure that a typo doesn't lead to moving the instance somewhere that would cause it to be immediately destroyed as an "orphan" on the next apply. That'd be consistent with our rule that you can't `terraform import` into a resource that isn't present in the configuration, so maybe there's some existing logic over there we could borrow and generalize to ensure they both follow the same rules.

Reply:
We actually don't handle config or prevent any of these sorts of things at all at the moment, so I think that's a question for a much larger change. Seeing how the normal workflow is to verify the change with a plan, I don't think typos will be too disastrous. Allowing individual instances to be moved also makes fixing any typos much easier as well.